pkg/cloudprovider/identity.go

```diff
@@ -29,3 +31,15 @@ func checkIfIDInList(idList []string, desiredID string) bool {
 	}
 	return false
 }
+
+// getResourceIdentityType returns resource identity type based on current type
+func getResourceIdentityType(identityType compute.ResourceIdentityType) compute.ResourceIdentityType {
```

Comment:
Could you please elaborate in the comment how the flow of code brings us to the state where we have to check the updated resource type based on the current resource type. Does function name like - getUpdatedIdentityType or something like that which indicates the nature of function make more sense ?